### PR TITLE
Separate ParachainId injection to its own pallet

### PR DESCRIPTION
**Pull request by JoshOrndorff**
This PR separates the task of injecting a ParachainId from the chainspec into storage from the message passing example pallet. This separation makes sense because the ParachainId injection must happen once in a runtime to get message receiving working, and should remain as part of the runtime crate. There may be multiple pallets that want to receive messages in a single runtime, and they may want to live in their own crates.

This is another (better) attempt at what I tried in #179.

### CODE REVIEW

**Comment by riusricardo** [comment]
If this is approved, I could be good to move the XCMPMessage to the message-example pallet because it is something specific of its behavior.
https://github.com/paritytech/cumulus/blob/master/test/parachain/runtime/src/lib.rs#L128

**Comment by JoshOrndorff** [comment]
F other stuff got merged in the meantime? Standby for another merge.

**Comment by bkchr** [comment]
> F other stuff got merged in the meantime? Standby for another merge.

Sorry

File: Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,9 @@ members = [
 	"parachain-upgrade",
 	"primitives",
 	"runtime",
+	"rococo-parachains/pallets/token-dealer",
+	"rococo-parachains/runtime",
```
Comment (bkchr) [inline Cargo.toml:10]: Look some lines below :P

File: rococo-parachains/runtime/src/parachain_info.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+decl_storage! {
+	trait Store for Module<T: Trait> as ParachainUpgrade {
+		ParachainId get(fn parachain_id) config(): ParaId = 100.into();
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin, system = frame_system {
+
+	}
```
Comment (bkchr) [inline rococo-parachains/runtime/src/parachain_info.rs:41]:
```suggestion
	pub struct Module<T: Trait> for enum Call where origin: T::Origin, system = frame_system {}
```

File: rococo-parachains/pallets/token-dealer/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+[package]
+authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2018"
+name = "token-dealer"
```
Comment (bkchr) [inline rococo-parachains/pallets/token-dealer/Cargo.toml:4]:
```suggestion
name = "cumulus-token-dealer"
```

File: rococo-parachains/pallets/token-dealer/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+version = "2.0.0-rc5"
```
Comment (bkchr) [inline rococo-parachains/pallets/token-dealer/Cargo.toml:5]:
```suggestion
version = "0.1.0"
```

File: rococo-parachains/contracts-runtime/src/parachain_info.rs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+decl_storage! {
+	trait Store for Module<T: Trait> as ParachainUpgrade {
+		ParachainId get(fn parachain_id) config(): ParaId = 100.into();
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin, system = frame_system {
+
+	}
```
Comment (bkchr) [inline rococo-parachains/contracts-runtime/src/parachain_info.rs:41]:
```suggestion
	pub struct Module<T: Trait> for enum Call where origin: T::Origin, system = frame_system {}
```

File: rococo-parachains/contracts-runtime/src/parachain_info.rs [hunk 6]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
```
Comment (bkchr) [inline rococo-parachains/contracts-runtime/src/parachain_info.rs:1]: Please make this also its own pallet.

File: Cargo.toml (hunk 1 above)
Comment (JoshOrndorff) [reply Cargo.toml:10]: :facepalm:

**Pull request merged**